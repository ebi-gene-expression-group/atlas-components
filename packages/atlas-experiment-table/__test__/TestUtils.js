import React from 'react'
import styled from 'styled-components'

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const TableCellDiv = styled.div`
  font-size: 13px;
  font-family: Helvetica, Arial, FreeSans, "Liberation Sans", sans-serif;
`

const tableHeader = [
  { type: `sort`, title: `Loaded date`, width: 140, dataParam: `lastUpdate` },
  { type: `search`, title: `species`, width: 200, dataParam: `species` },
  { type: ``, title: `experiment description`, width: 360, dataParam: `experimentDescription`, link: `experimentAccession`, resource: `experiments`, endpoint: `Results` },
  { type: `plain`, title: `experiment factors`, width: 260, dataParam: `experimentalFactors` },
  { type: `sort`, title: `Number of assays`, width: 160, dataParam: `numberOfAssays`, link: `experimentAccession`, resource: `experiments`, endpoint: `Experiment Design` },
]

const data = [
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-EHCA-2`,
    experimentDescription: `Melanoma infiltration`,
    lastUpdate: `16-11-2018`,
    numberOfAssays: 6638,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`, `sampling site`,`time`],
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-GEOD-99058`,
    experimentDescription: `Single cell`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 250,
    numberOfContrasts: 0,
    species: `Mus musculus`,
    kingdom: `animals`,
    experimentalFactors: [`single cell identifier`]
  },
  {
    experimentType: `SINGLE`,
    experimentAccession: `E-MTAB-5061`,
    experimentDescription: `healthy individuals and type 2 diabetes patients`,
    lastUpdate: `11-10-2018`,
    numberOfAssays: 3514,
    numberOfContrasts: 0,
    species: `Homo sapiens`,
    kingdom: `plants`,
    experimentalFactors: [`single cell identifier`, `disease`],
  }
]

export {getRandomInt, TableCellDiv, tableHeader, data}
