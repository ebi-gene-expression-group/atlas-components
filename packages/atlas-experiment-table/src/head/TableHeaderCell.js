import React from 'react'
import PropTypes from 'prop-types'

import { Table, SearchInput, Heading, Pane, Icon, majorScale } from 'evergreen-ui'

const TableHeaderCell =
({
  label, dataKey,
  width, searchable, value, onChange, sortable, columnIndex, sortColumnIndex, ascendingOrder, onClick
}) =>
  <Table.HeaderCell
    flexGrow={width}
    cursor={sortable ? searchable ? `auto` : `pointer` : `auto`}>
    {
      searchable ?
        <SearchInput
          size={500}
          width={`100%`}
          placeholder={`${label}`}
          border={`muted`}
          value={value}
          onChange={e => onChange(dataKey, e.target.value)}/> :
        <Heading
          size={500}
          onClick={sortable ? () => onClick(columnIndex) : () => {}}>
          {label}
        </Heading>
    }
    {
      sortable &&
      <Pane paddingLeft={majorScale(1)}>
        <Icon
          icon={columnIndex === sortColumnIndex ? ascendingOrder ? `sort-asc` : `sort-desc` : `sort`}
          onClick={() => onClick(columnIndex)}
          cursor={`pointer`}/>
      </Pane>
    }
  </Table.HeaderCell>

TableHeaderCell.propTypes = {
  label: PropTypes.string.isRequired,
  // Even though dataKey isn’t used when a plain header cell is created, it’s necessary to have it or the corresponding
  // field won’t be shown for this column
  dataKey: PropTypes.string.isRequired,
  width: PropTypes.number,
  searchable: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func,
  sortable: PropTypes.bool,
  columnIndex: PropTypes.number,
  sortColumnIndex: PropTypes.number,
  ascendingOrder: PropTypes.bool,
  onClick: PropTypes.func
}

TableHeaderCell.defaultProps = {
  width: 1,
  searchable: false,
  value: null,
  onChange: () => {},
  sortable: false,
  columnIndex: 0,
  sortColumnIndex: 0,
  ascendingOrder: false,
  onClick: () => {}
}

export default TableHeaderCell
