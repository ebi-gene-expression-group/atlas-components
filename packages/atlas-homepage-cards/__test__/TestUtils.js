

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const dummyCards = () => {
  return [
    {
      iconType: `species`,
      iconSrc: `mus musculus`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `species`,
      iconSrc: `homo sapiens`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `unknown`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
    {
      iconType: `experiments`,
      iconSrc: `http://www.zhao.com`,
      iconDescription: `Homo sapiens`,
      content: [
        {
          text: `2 experiments`,
        }
      ]
    },
    {
      iconType: `image`,
      iconSrc: `logo-goes-here`,
      iconDescription: `An image card`,
      content: [
        {
          text: `Go to experiments`,
        }
      ]
    },
    {
      iconType: `imagespecies`,
      iconSrc: `http://www.lingyun.com`,
      iconDescription: `Mus musculus`,
      content: [
        {
          text: `4 experiments`,
        }
      ]
    },
  ]
}

export {getRandomInt, dummyCards}