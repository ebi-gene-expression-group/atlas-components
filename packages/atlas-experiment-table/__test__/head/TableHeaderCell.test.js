import React from 'react'
import { shallow, mount } from 'enzyme'
import TableHeaderCell from '../../src/head/TableHeaderCell'

import { Table, Heading, SearchInput, Icon } from 'evergreen-ui'

import { getRandomInt } from '../TestUtils'
import randomString from 'random-string'

describe(`TableHeaderCell`, () => {
  const MAX_NUMBER_OF_COLUMNS = 10

  test(`displays a table header cell with only a label if only the required props are passed`, () => {
    const props = {
      label: randomString(),
      dataKey: randomString()
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    expect(wrapper.find(SearchInput)).toHaveLength(0)
    expect(wrapper.find(Icon)).toHaveLength(0)

    expect(wrapper.find(Table.HeaderCell)).toHaveLength(1)
    expect(wrapper.find(Heading)).toHaveLength(1)

    expect(wrapper.find(Heading)).toHaveProp(`children`, props.label)
  })

  test(`displays a text input if it’s searchable`, () => {
    const props = {
      label: randomString(),
      dataKey: randomString(),
      searchable: true
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    expect(wrapper.find(Icon)).toHaveLength(0)

    expect(wrapper.find(SearchInput)).toHaveLength(1)
    expect(wrapper.find(SearchInput)).toHaveProp(`placeholder`, props.label)
  })

  test(`can have an initial value if it’s searchable`, () => {
    const props = {
      label: randomString(),
      dataKey: randomString(),
      searchable: true,
      value: randomString()
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    expect(wrapper.find(SearchInput)).toHaveProp(`value`, props.value)
  })

  test(`displays a sort icon if it is marked as sortable`, () => {
    const props = {
      label: randomString(),
      dataKey: randomString(),
      sortable: true
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    expect(wrapper.find(SearchInput)).toHaveLength(0)

    expect(wrapper.find(Icon)).toHaveLength(1)
    expect(wrapper.find(Heading)).toHaveLength(1)

    expect(wrapper.find(Heading)).toHaveProp(`children`, props.label)
  })

  test(`calls onChange if the search input is changed`, () => {
    const props = {
      label: randomString(),
      dataKey: randomString(),
      searchable: true,
      onChange: jest.fn()
    }
    const event = { target: { value: randomString() } }
    const wrapper = shallow(<TableHeaderCell {...props}/>)
    wrapper.find(SearchInput).simulate(`change`, event)

    expect(props.onChange).toHaveBeenCalled()
    expect(props.onChange.mock.calls).toContainEqual([props.dataKey, event.target.value])
  })

  test(`displays an ascending/descending sort icon if this header cell dictates the sort order`, () => {
    const randomColumnIndex = getRandomInt(0, MAX_NUMBER_OF_COLUMNS)
    const props = {
      label: randomString(),
      dataKey: randomString(),
      sortable: true,
      onClick: jest.fn(),
      columnIndex: randomColumnIndex,
      sortColumnIndex: randomColumnIndex,
      ascendingOrder: Math.random() < 0.5
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    expect(wrapper.find(Icon)).toHaveProp(`icon`, props.ascendingOrder ? `sort-asc` : `sort-desc`)
  })

  test(`calls onClick when the icon of a sortable header cell is clicked`, () => {
    const randomColumnIndex = getRandomInt(0, MAX_NUMBER_OF_COLUMNS)
    const props = {
      label: randomString(),
      dataKey: randomString(),
      searchable: Math.random() < 0.5,
      sortable: true,
      onClick: jest.fn(),
      columnIndex: randomColumnIndex,
      sortColumnIndex: randomColumnIndex + 1,
      ascendingOrder: Math.random() < 0.5
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)

    wrapper.find(Icon).simulate(`click`)
    expect(props.onClick).toHaveBeenCalled()
  })

  test(`calls onClick when the label of a non-searchable sortable header cell is clicked`, () => {
    const randomColumnIndex = getRandomInt(0, MAX_NUMBER_OF_COLUMNS)
    const props = {
      label: randomString(),
      dataKey: randomString(),
      sortable: true,
      onClick: jest.fn(),
      columnIndex: randomColumnIndex,
      sortColumnIndex: randomColumnIndex,
      ascendingOrder: Math.random() < 0.5
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)
    wrapper.find(Heading).simulate(`click`)

    expect(props.onClick).toHaveBeenCalled()
  })

  test(`doesn’t call onClick when the text input in a sortable header cell is clicked`, () => {
    const randomColumnIndex = getRandomInt(0, MAX_NUMBER_OF_COLUMNS)
    const props = {
      label: randomString(),
      dataKey: randomString(),
      searchable: true,
      sortable: true,
      onClick: jest.fn(),
      columnIndex: randomColumnIndex,
      sortColumnIndex: randomColumnIndex,
      ascendingOrder: Math.random() < 0.5
    }
    const wrapper = shallow(<TableHeaderCell {...props}/>)
    wrapper.find(SearchInput).simulate(`click`)

    expect(props.onClick).not.toHaveBeenCalled()
  })

  test(`matches snapshot (basic)`, () => {
    const props = {
      label: `Header cell label`,
      dataKey: `Data key field`
    }
    expect(mount(<TableHeaderCell {...props}/>)).toMatchSnapshot()
  })

  test(`matches snapshot (searchable)`, () => {
    const props = {
      label: `Header cell label`,
      dataKey: `Data key field`,
      searchable: true,
      value: ``,
      onChange: jest.fn()}
    expect(mount(<TableHeaderCell {...props}/>)).toMatchSnapshot()
  })

  test(`matches snapshot (sortable)`, () => {
    const props = {
      label: `Header cell label`,
      dataKey: `Data key field`,
      sortable: true,
      columnIndex: 0,
      sortColumnIndex: 0,
      ascendingOrder: true,
      onClick: jest.fn()
    }
    expect(mount(<TableHeaderCell {...props}/>)).toMatchSnapshot()
  })

  test(`matches snapshot (searchable and sortable)`, () => {
    const props = {
      label: `Header cell label`,
      dataKey: `Data key field`,
      searchable: true,
      value: ``,
      onChange: jest.fn(),
      sortable: true,
      columnIndex: 0,
      sortColumnIndex: 0,
      ascendingOrder: true,
      onClick: jest.fn()
    }
    expect(mount(<TableHeaderCell {...props}/>)).toMatchSnapshot()
  })
})
