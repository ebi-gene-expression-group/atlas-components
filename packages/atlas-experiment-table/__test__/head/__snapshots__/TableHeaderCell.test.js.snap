// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeaderCell matches snapshot (basic) 1`] = `
<TableHeaderCell
  ascendingOrder={false}
  columnIndex={0}
  dataKey="Data key field"
  label="Header cell label"
  onChange={[Function]}
  onClick={[Function]}
  searchable={false}
  sortColumnIndex={0}
  sortable={false}
  value={null}
  width={1}
>
  <Memo(TableHeaderCell)
    cursor="auto"
    flexGrow={1}
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      cursor="auto"
      flexGrow={1}
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        cursor="auto"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          cursor="auto"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-crsr_auto"
            onKeyDown={[Function]}
          >
            <ForwardRef(Heading)
              onClick={[Function]}
              size={500}
            >
              <Box
                boxSizing="border-box"
                color="#234361"
                fontFamily="\\"SF UI Text\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\""
                fontSize="16px"
                fontWeight={500}
                is="h2"
                letterSpacing="-0.05px"
                lineHeight="20px"
                marginBottom={0}
                marginTop={0}
                onClick={[Function]}
              >
                <h2
                  className="ub-mt_0px ub-mb_0px ub-fnt-sze_16px ub-f-wght_500 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_234361 ub-box-szg_border-box"
                  onClick={[Function]}
                >
                  Header cell label
                </h2>
              </Box>
            </ForwardRef(Heading)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</TableHeaderCell>
`;

exports[`TableHeaderCell matches snapshot (searchable and sortable) 1`] = `
<TableHeaderCell
  ascendingOrder={true}
  columnIndex={0}
  dataKey="Data key field"
  label="Header cell label"
  onChange={[MockFunction]}
  onClick={[MockFunction]}
  searchable={true}
  sortColumnIndex={0}
  sortable={true}
  value=""
  width={1}
>
  <Memo(TableHeaderCell)
    cursor="auto"
    flexGrow={1}
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      cursor="auto"
      flexGrow={1}
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        cursor="auto"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          cursor="auto"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-crsr_auto"
            onKeyDown={[Function]}
          >
            <ForwardRef(SearchInput)
              border="muted"
              onChange={[Function]}
              placeholder="Header cell label"
              size={500}
              value=""
              width="100%"
            >
              <Box
                border="muted"
                boxSizing="border-box"
                display="inline-flex"
                height={32}
                is="div"
                position="relative"
                width="100%"
              >
                <div
                  className="ub-pst_relative ub-dspl_inline-flex ub-h_32px ub-w_100prcnt ub-b-btm_muted ub-b-lft_muted ub-b-rgt_muted ub-b-top_muted ub-box-szg_border-box"
                >
                  <Box
                    alignItems="center"
                    boxSizing="border-box"
                    display="flex"
                    height={32}
                    is="div"
                    justifyContent="center"
                    pointerEvents="none"
                    position="absolute"
                    width={32}
                  >
                    <div
                      className="ub-h_32px ub-w_32px ub-ptr-evts_none ub-pst_absolute ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-box-szg_border-box"
                    >
                      <ForwardRef(SearchIcon)
                        color="default"
                        size={12}
                        zIndex={3}
                      >
                        <ForwardRef(Icon)
                          color="default"
                          name="search"
                          size={12}
                          svgPaths16={
                            Array [
                              "M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z",
                            ]
                          }
                          svgPaths20={
                            Array [
                              "M19.56 17.44l-4.94-4.94A8.004 8.004 0 0016 8c0-4.42-3.58-8-8-8S0 3.58 0 8s3.58 8 8 8c1.67 0 3.21-.51 4.5-1.38l4.94 4.94a1.498 1.498 0 102.12-2.12zM8 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z",
                            ]
                          }
                          zIndex={3}
                        >
                          <Box
                            boxSizing="border-box"
                            data-icon="search"
                            height={12}
                            is="svg"
                            style={
                              Object {
                                "fill": "#66788A",
                              }
                            }
                            viewBox="0 0 16 16"
                            width={12}
                            zIndex={3}
                          >
                            <svg
                              className="ub-z-idx_3 ub-w_12px ub-h_12px ub-box-szg_border-box"
                              data-icon="search"
                              style={
                                Object {
                                  "fill": "#66788A",
                                }
                              }
                              viewBox="0 0 16 16"
                            >
                              <path
                                d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                                fillRule="evenodd"
                                key="0"
                              />
                            </svg>
                          </Box>
                        </ForwardRef(Icon)>
                      </ForwardRef(SearchIcon)>
                    </div>
                  </Box>
                  <ForwardRef(TextInput)
                    appearance="default"
                    height={32}
                    onChange={[Function]}
                    paddingLeft={32}
                    placeholder="Header cell label"
                    size={500}
                    type="search"
                    value=""
                    width="100%"
                  >
                    <ForwardRef(Text)
                      aria-invalid={false}
                      borderRadius={3}
                      className="css-5ljhhe"
                      disabled={false}
                      height={32}
                      is="input"
                      onChange={[Function]}
                      paddingLeft={32}
                      paddingRight={10}
                      placeholder="Header cell label"
                      size={500}
                      spellCheck={true}
                      type="search"
                      value=""
                      width="100%"
                    >
                      <Box
                        aria-invalid={false}
                        borderRadius={3}
                        boxSizing="border-box"
                        className="css-5ljhhe"
                        color="#425A70"
                        disabled={false}
                        fontFamily="\\"SF UI Text\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\""
                        fontSize="16px"
                        fontWeight={400}
                        height={32}
                        is="input"
                        letterSpacing="-0.05px"
                        lineHeight="20px"
                        onChange={[Function]}
                        paddingLeft={32}
                        paddingRight={10}
                        placeholder="Header cell label"
                        spellCheck={true}
                        type="search"
                        value=""
                        width="100%"
                      >
                        <input
                          aria-invalid={false}
                          className="css-5ljhhe ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-w_100prcnt ub-h_32px ub-pl_32px ub-pr_10px ub-bblr_3px ub-bbrr_3px ub-btlr_3px ub-btrr_3px ub-box-szg_border-box"
                          disabled={false}
                          onChange={[Function]}
                          placeholder="Header cell label"
                          spellCheck={true}
                          type="search"
                          value=""
                        />
                      </Box>
                    </ForwardRef(Text)>
                  </ForwardRef(TextInput)>
                </div>
              </Box>
            </ForwardRef(SearchInput)>
            <ForwardRef(Pane)
              paddingLeft={8}
            >
              <Box
                boxSizing="border-box"
                className="css-nil"
                is="div"
                paddingLeft={8}
              >
                <div
                  className="css-nil ub-pl_8px ub-box-szg_border-box"
                >
                  <ForwardRef(Icon)
                    cursor="pointer"
                    icon="sort-asc"
                    onClick={[Function]}
                  />
                </div>
              </Box>
            </ForwardRef(Pane)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</TableHeaderCell>
`;

exports[`TableHeaderCell matches snapshot (searchable) 1`] = `
<TableHeaderCell
  ascendingOrder={false}
  columnIndex={0}
  dataKey="Data key field"
  label="Header cell label"
  onChange={[MockFunction]}
  onClick={[Function]}
  searchable={true}
  sortColumnIndex={0}
  sortable={false}
  value=""
  width={1}
>
  <Memo(TableHeaderCell)
    cursor="auto"
    flexGrow={1}
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      cursor="auto"
      flexGrow={1}
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        cursor="auto"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          cursor="auto"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-crsr_auto"
            onKeyDown={[Function]}
          >
            <ForwardRef(SearchInput)
              border="muted"
              onChange={[Function]}
              placeholder="Header cell label"
              size={500}
              value=""
              width="100%"
            >
              <Box
                border="muted"
                boxSizing="border-box"
                display="inline-flex"
                height={32}
                is="div"
                position="relative"
                width="100%"
              >
                <div
                  className="ub-pst_relative ub-dspl_inline-flex ub-h_32px ub-w_100prcnt ub-b-btm_muted ub-b-lft_muted ub-b-rgt_muted ub-b-top_muted ub-box-szg_border-box"
                >
                  <Box
                    alignItems="center"
                    boxSizing="border-box"
                    display="flex"
                    height={32}
                    is="div"
                    justifyContent="center"
                    pointerEvents="none"
                    position="absolute"
                    width={32}
                  >
                    <div
                      className="ub-h_32px ub-w_32px ub-ptr-evts_none ub-pst_absolute ub-dspl_flex ub-just-cnt_center ub-algn-itms_center ub-box-szg_border-box"
                    >
                      <ForwardRef(SearchIcon)
                        color="default"
                        size={12}
                        zIndex={3}
                      >
                        <ForwardRef(Icon)
                          color="default"
                          name="search"
                          size={12}
                          svgPaths16={
                            Array [
                              "M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z",
                            ]
                          }
                          svgPaths20={
                            Array [
                              "M19.56 17.44l-4.94-4.94A8.004 8.004 0 0016 8c0-4.42-3.58-8-8-8S0 3.58 0 8s3.58 8 8 8c1.67 0 3.21-.51 4.5-1.38l4.94 4.94a1.498 1.498 0 102.12-2.12zM8 14c-3.31 0-6-2.69-6-6s2.69-6 6-6 6 2.69 6 6-2.69 6-6 6z",
                            ]
                          }
                          zIndex={3}
                        >
                          <Box
                            boxSizing="border-box"
                            data-icon="search"
                            height={12}
                            is="svg"
                            style={
                              Object {
                                "fill": "#66788A",
                              }
                            }
                            viewBox="0 0 16 16"
                            width={12}
                            zIndex={3}
                          >
                            <svg
                              className="ub-z-idx_3 ub-w_12px ub-h_12px ub-box-szg_border-box"
                              data-icon="search"
                              style={
                                Object {
                                  "fill": "#66788A",
                                }
                              }
                              viewBox="0 0 16 16"
                            >
                              <path
                                d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                                fillRule="evenodd"
                                key="0"
                              />
                            </svg>
                          </Box>
                        </ForwardRef(Icon)>
                      </ForwardRef(SearchIcon)>
                    </div>
                  </Box>
                  <ForwardRef(TextInput)
                    appearance="default"
                    height={32}
                    onChange={[Function]}
                    paddingLeft={32}
                    placeholder="Header cell label"
                    size={500}
                    type="search"
                    value=""
                    width="100%"
                  >
                    <ForwardRef(Text)
                      aria-invalid={false}
                      borderRadius={3}
                      className="css-5ljhhe"
                      disabled={false}
                      height={32}
                      is="input"
                      onChange={[Function]}
                      paddingLeft={32}
                      paddingRight={10}
                      placeholder="Header cell label"
                      size={500}
                      spellCheck={true}
                      type="search"
                      value=""
                      width="100%"
                    >
                      <Box
                        aria-invalid={false}
                        borderRadius={3}
                        boxSizing="border-box"
                        className="css-5ljhhe"
                        color="#425A70"
                        disabled={false}
                        fontFamily="\\"SF UI Text\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\""
                        fontSize="16px"
                        fontWeight={400}
                        height={32}
                        is="input"
                        letterSpacing="-0.05px"
                        lineHeight="20px"
                        onChange={[Function]}
                        paddingLeft={32}
                        paddingRight={10}
                        placeholder="Header cell label"
                        spellCheck={true}
                        type="search"
                        value=""
                        width="100%"
                      >
                        <input
                          aria-invalid={false}
                          className="css-5ljhhe ub-color_425A70 ub-fnt-fam_b77syt ub-fnt-sze_16px ub-f-wght_400 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-w_100prcnt ub-h_32px ub-pl_32px ub-pr_10px ub-bblr_3px ub-bbrr_3px ub-btlr_3px ub-btrr_3px ub-box-szg_border-box"
                          disabled={false}
                          onChange={[Function]}
                          placeholder="Header cell label"
                          spellCheck={true}
                          type="search"
                          value=""
                        />
                      </Box>
                    </ForwardRef(Text)>
                  </ForwardRef(TextInput)>
                </div>
              </Box>
            </ForwardRef(SearchInput)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</TableHeaderCell>
`;

exports[`TableHeaderCell matches snapshot (sortable) 1`] = `
<TableHeaderCell
  ascendingOrder={true}
  columnIndex={0}
  dataKey="Data key field"
  label="Header cell label"
  onChange={[Function]}
  onClick={[MockFunction]}
  searchable={false}
  sortColumnIndex={0}
  sortable={true}
  value={null}
  width={1}
>
  <Memo(TableHeaderCell)
    cursor="pointer"
    flexGrow={1}
  >
    <ForwardRef(TableCell)
      borderBottom={null}
      cursor="pointer"
      flexGrow={1}
      overflow="visible"
    >
      <ForwardRef(Pane)
        alignItems="center"
        borderBottom={null}
        boxSizing="border-box"
        className="css-saeo60"
        cursor="pointer"
        display="flex"
        flex={1}
        flexGrow={1}
        flexShrink={0}
        onKeyDown={[Function]}
        overflow="visible"
        paddingX={12}
      >
        <Box
          alignItems="center"
          borderBottom={null}
          boxSizing="border-box"
          className="css-saeo60 css-nil"
          cursor="pointer"
          display="flex"
          flex={1}
          flexGrow={1}
          flexShrink={0}
          is="div"
          onKeyDown={[Function]}
          overflow="visible"
          paddingX={12}
        >
          <div
            className="css-saeo60 css-nil ub-pl_12px ub-pr_12px ub-box-szg_border-box ub-flx_1 ub-dspl_flex ub-algn-itms_center ub-flx-srnk_0 ub-ovflw-x_visible ub-ovflw-y_visible ub-flx-grow_1 ub-crsr_pointer"
            onKeyDown={[Function]}
          >
            <ForwardRef(Heading)
              onClick={[Function]}
              size={500}
            >
              <Box
                boxSizing="border-box"
                color="#234361"
                fontFamily="\\"SF UI Text\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\""
                fontSize="16px"
                fontWeight={500}
                is="h2"
                letterSpacing="-0.05px"
                lineHeight="20px"
                marginBottom={0}
                marginTop={0}
                onClick={[Function]}
              >
                <h2
                  className="ub-mt_0px ub-mb_0px ub-fnt-sze_16px ub-f-wght_500 ub-ln-ht_20px ub-ltr-spc_-0-05px ub-fnt-fam_b77syt ub-color_234361 ub-box-szg_border-box"
                  onClick={[Function]}
                >
                  Header cell label
                </h2>
              </Box>
            </ForwardRef(Heading)>
            <ForwardRef(Pane)
              paddingLeft={8}
            >
              <Box
                boxSizing="border-box"
                className="css-nil"
                is="div"
                paddingLeft={8}
              >
                <div
                  className="css-nil ub-pl_8px ub-box-szg_border-box"
                >
                  <ForwardRef(Icon)
                    cursor="pointer"
                    icon="sort-asc"
                    onClick={[Function]}
                  />
                </div>
              </Box>
            </ForwardRef(Pane)>
          </div>
        </Box>
      </ForwardRef(Pane)>
    </ForwardRef(TableCell)>
  </Memo(TableHeaderCell)>
</TableHeaderCell>
`;
